html/includes/api_functions.inc.php

```diff
@@ -507,6 +508,59 @@ function get_graph_by_portgroup() {
 }
 
 
+function get_components() {
+    global $config;
+    $code     = 200;
+    $status   = 'ok';
+    $message  = '';
+    $app      = \Slim\Slim::getInstance();
+    $router   = $app->router()->getCurrentRoute()->getParams();
+    $hostname = $router['hostname'];
+
+    // Do some filtering if the user requests.
```

Comment:
This section feels like it could be collapsed into one data structure + a loop.  Not a blocker, though.